backend-rust/migrations/0001_initialize.up.sql

```diff
@@ -0,0 +1,224 @@
+    -- This is NULL until the indexer have processed the block marking this a finalized.
+    finalization_time
+        INTEGER,
+    -- Block causing this block to become finalized.
+    -- This is NULL until the indexer have processed the block marking this a finalized.
+    finalized_by
+        BIGINT
+        REFERENCES blocks(height),
+    -- Index of the account which baked the block.
+    -- For non-genesis blocks this should always be defined.
+    -- Foreign key constraint added later, since account table is not defined yet.
+    baker_id
+        BIGINT,
+    -- The total amount of CCD in existence at the time of this block was created in micro CCD.
+    total_amount
+        BIGINT
+        NOT NULL,
+    -- The total staked amount of CCD at the time of this block was created in micro CCD.
+    total_staked
+        BIGINT
+        NOT NULL
+);
+
+-- Every transaction on chain.
+CREATE TABLE transactions(
+    -- Index of the transaction within the block.
+    index
+        BIGINT
+        NOT NULL,
+    -- Absolute height of the block containing the transaction.
+    block
+        BIGINT
+        REFERENCES blocks(height)
+        NOT NULL,
+    -- Transaction hash encoded using HEX.
+    hash
+        CHAR(64)
+        UNIQUE
+        NOT NULL,
+    -- The cost of the transaction in terms of CCD.
+    ccd_cost
+        BIGINT
+        NOT NULL,
+    -- The energy cost of the transaction.
+    energy_cost
+        BIGINT
+        NOT NULL,
+    -- The account used for sending of the transaction.
+    -- NULL for chain update and account creation transactions.
+    -- Foreign key constraint added later, since account table is not defined yet.
+    sender
+        BIGINT,
+    -- The type of transaction.
+    type
+        transaction_type
+        NOT NULL,
+    -- NULL if the transaction type is not account or the account transaction have no effect on chain.
+    type_account
+        account_transaction_type,
+    -- NULL if the transaction type is not credential deployment.
+    type_credential_deployment
+        credential_deployment_transaction_type,
+    -- NULL if the transaction type is not update.
+    type_update
+        update_transaction_type,
+    -- Whether the transaction was accepted or rejected.
+    success
+        BOOLEAN
+        NOT NULL,
+    -- Transaction details. Events if success is true.
+    events
+        JSONB,
+    -- Transaction details. Reject reason if success is false.
+    reject
+        JSONB,
+
+    -- Make the block height and transaction index the primary key.
+    PRIMARY KEY (block, index)
+    -- transaction_type: TransactionType,
+);
+
+-- Every account on chain.
+CREATE TABLE accounts(
+    -- Index of the account.
+    index
+        BIGINT
+        PRIMARY KEY
+        NOT NULL,
+    -- Account address bytes encoded using base58check.
+    address
+        CHAR(50)
+        UNIQUE
+        NOT NULL,
```

Comment:
Good point.
I initially wanted to put the byte representation in, but since the API provides search for an account address by some prefix, I ended up putting the base58check representation in.